### ISSUE TITLE
OAuth2 endpoint redirects to localhost instead of public IP

### ISSUE THREAD

**Issue by wleev** [Erudika/para#53]
Generic OAuth2 filter worked on my localhost dev environment, but seems to have a small bug when deployed to a test server with proper domain names.

I have set up Nginx reverse proxies for both Para and Scoold, which seem to work fine.
Except for the `oauth2_auth` endpoint which redirects to `localhost` instead of the expected public domain and makes the authentication flow fail. ( redirect looking like this: `http://localhost:8040/signin/success?jwt=<token>` instead of `http://qa.test.isabot.net/signin/success?jwt=<token>` )

Scoold's application.conf has both URL's set up correctly:
```
para.host_url = "http://qa.test.isabot.net"
para.endpoint = "http://para.test.isabot.net"
```

Am I missing something in Para's application.conf?

**Comment by albogdano** [comment]
Do you have `para.signin_success = "success_url"` and `para.signin_failure = "fail_url"` in your Scoold config file?

**Comment by wleev** [comment]
I don't actually, I didn't see them in the example config.
Should I copy them in like that or use a more specific value like for instance "http://qa.test.isabot.net/signin/success`?

**Comment by albogdano** [comment]
No, actually they are not required to be in the config file because they are constructed from the `host_url`. One thing to check is that you have `para.env = "production"` this is quite important because it switches between `localhost` and the value of `{host_url}` for those redirects.

**Comment by albogdano** [comment]
Could it be that the email for that user is empty/blank? That warning is only shown if `para.approved_domains_for_signups` is set *or* if the email is blank (which is not allowed).

**Comment by albogdano** [comment]
Oops, that's my bad! :sweat_smile:  It should be fixed now. I'm releasing a new patch right away.

**Comment by wleev** [comment]
Great! I'll give it a test right away.

**Comment by albogdano** [comment]
Just to clarify, the bug was in Scoold - [v1.30.5](https://github.com/Erudika/scoold/releases/tag/1.30.5) is now released.

**Comment by wleev** [comment]
Confirmed working.
